experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1258 @@
+end
+
+### Differential ###
+@doc raw"""
+    differential_polynomial_ring(R::Ring, n_elementary_symbols::Int, ndiffs::Int) -> Tuple{DifferentialPolyRing, Vector{DifferentialPolyRingElem}}
+
+Return a tuple consisting of the differential polynomial ring over the ring `R` with the specified number of elementary variables and commuting endomorphisms, and the vector of
+these elementary variables. This methods allows for all the keywords of the `set_ranking!` method. See there for further information.
+
+# Examples
+
+```jldoctests
+julia> S, variables = differential_polynomial_ring(QQ, [:a, :b, :c], 4)
+(Differential polynomial ring in 3 elementary symbols over QQ, DifferentialPolyRingElem{QQFieldElem}[a[0,0,0,0], b[0,0,0,0], c[0,0,0,0]])
+
+julia> S
+Differential polynomial ring in 3 elementary symbols a, b, c
+with 4 commuting derivations
+  over rational field
+
+julia> variables
+3-element Vector{DifferentialPolyRingElem{QQFieldElem}}:
+ a[0,0,0,0]
+ b[0,0,0,0]
+ c[0,0,0,0]
+```
+"""
+function differential_polynomial_ring(R::Ring, n_elementary_symbols::Int, ndiffs::Int; kwargs...)
+  dpr = DifferentialPolyRing{elem_type(typeof(R))}(R, n_elementary_symbols, ndiffs)
+  set_ranking!(dpr; kwargs...)
+  return (dpr, deepcopy.(__add_new_jetvar!(dpr, [(i, zeros(Int, ndiffs)) for i in 1:n_elementary_symbols])))
+end
+
+@doc raw"""
+    differential_polynomial_ring(R::Ring, elementary_symbols::Vector{Symbol}, ndiffs::Int) -> Tuple{DifferentialPolyRing, Vector{DifferentialPolyRingElem}}
+
+Return a tuple consisting of the differential polynomial ring over the ring `R` with the specified elementary variables and number of commuting endomorphisms, and the vector of
+these elementary variables. Note that the multiindex [0..0] of length 'ndiffs' is appended to the variable names provided. This methods allows for all the keywords of the `set_ranking!`
+method. See there for further information.
+
+# Examples
+
+```jldoctests
+julia> S, variables = differential_polynomial_ring(QQ, 3, 4)
+(Differential polynomial ring in 3 elementary symbols over QQ, DifferentialPolyRingElem{QQFieldElem}[u1[0,0,0,0], u2[0,0,0,0], u3[0,0,0,0]])
+
+julia> S
+Differential polynomial ring in 3 elementary symbols u1, u2, u3
+with 4 commuting derivations
+  over rational field
+
+julia> variables
+3-element Vector{DifferentialPolyRingElem{QQFieldElem}}:
+ u1[0,0,0,0]
+ u2[0,0,0,0]
+ u3[0,0,0,0]
+```
+"""
+function differential_polynomial_ring(R::Ring, elementary_symbols::Vector{Symbol}, ndiffs::Int; kwargs...)
+  dpr = DifferentialPolyRing{elem_type(typeof(R))}(R, elementary_symbols, ndiffs)
+  set_ranking!(dpr; kwargs...)
+  return (dpr, deepcopy.(__add_new_jetvar!(dpr, [(i, zeros(Int, ndiffs)) for i in 1:length(elementary_symbols)])))
+end
+
+##### Elements #####
+
+### Difference ###
+(dpr::DifferencePolyRing)() = DifferencePolyRingElem{elem_type(base_ring(dpr))}(dpr)
```

Comment:
```suggestion
(dpr::DifferencePolyRing)() = elem_type(dpr)(dpr)
```
and similar for the other ones directly below. With this change, you can then again merge them to a union signature